crates/driver/src/domain/competition/mod.rs

```diff
@@ -400,9 +400,23 @@ impl Competition {
                 }
 
                 observe::settling();
-                let result = self
-                    .process_settle_request(auction_id, solution_id, submission_deadline)
-                    .await;
+                let cancel_signal = Arc::new(tokio::sync::Notify::new());
+                let cancel_signal_clone = cancel_signal.clone();
+                let settle_fut = self.process_settle_request(
+                    auction_id,
+                    solution_id,
+                    submission_deadline,
+                    cancel_signal_clone,
+                );
+                let result = tokio::select! {
+                    // Check whether the sender is closed, which indicates that the client
+                    // has disconnected and the settlement task needs to be cancelled.
+                    _ = response_sender.closed() => {
+                        cancel_signal.notify_waiters();
+                        Err(DeadlineExceeded.into())
```

Comment:
IIUC simply dropping and stopping to poll the submission future would be equivalent to what's happening now. So if we want to go down that route I think it would be simpler to just do that.

And actually I think this might already be happening unintentionally. The `select!` only waits for one of the tasks to finish. If it's the channel close future we would return an error here and the settle future would not get polled anymore. That means the settle future wouldn't even realize that a cancellation signal happened.